src/sentry/static/sentry/app/components/button.tsx

```diff
@@ -163,7 +163,7 @@ class Button extends React.Component<ButtonProps, {}> {
           priority={priority}
           borderless={borderless}
         >
-          {icon && (
+          {icon && typeof icon === 'string' ? (
```

Comment:
Ah, looking at the snapshots, it looks like the "else" condition will pick up the case where icon is falsey, we could move the `typeof icon` check to inside of `<Icon>`